tiled/client/context.py

```diff
@@ -159,6 +159,23 @@ def __init__(
         self.api_key = api_key  # property setter sets Authorization header
         self.admin = Admin(self)  # accessor for admin-related requests
 
+    def __repr__(self):
+        auth_info = []
+        if (self.api_key is None) and (self.http_client.auth is None):
+            auth_info.append("(unauthenticated)")
+        else:
+            auth_info.append("authenticated")
+            identities = self.whoami()["identities"]
+            if identities:
+                auth_info.append("as")
+                auth_info.append(
+                    ",".join(f"'{identity['id']}'" for identity in identities)
+                )
+            if self.api_key is not None:
+                auth_info.append(f"with API key '{self.api_key[:8]}...'")
```

Comment:
Presumably API keys are a respectable length, but perhaps just to guard against short keys being leaked in error reports that people post...

```suggestion
                auth_info.append(f"with API key '{self.api_key[:min(len(self.api_key)//2, 8)]}...'")
```

Reply:
Fair point.

Just for info for anyone following along, in a multi-user server, the API keys are only ever generated by the user and are always 32 bits of entropy. It's only in a single-user server context, where the user specified an API key, that this would come up.